scipy/stats/_qmc.py

```diff
@@ -164,7 +164,7 @@ def scale(
 
     if not reverse:
         # Checking that sample is within the hypercube
-        if not (np.all(sample >= 0) and np.all(sample <= 1)):
+        if not ((sample.max() <= 1.) and (sample.min() >= 0.)):
```

Comment:
Can't review without making suggestions, right?

```suggestion
        if (sample.max() > 1.) or (sample.min() < 0.):
```

Reply:
Haha sure!

Indeed, better than having the not.

Reply:
Shall I wait for this to merge or go for it?